llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -356,7 +356,8 @@ bool MachineLICMBase::runOnMachineFunction(MachineFunction &MF) {
   MRI = &MF.getRegInfo();
   SchedModel.init(&ST);
 
-  PreRegAlloc = MRI->isSSA();
+  PreRegAlloc = !MF.getProperties().hasProperty(
+      MachineFunctionProperties::Property::NoVRegs);
```

Comment:
Tbh one should never redefine `PreRegAlloc`, there's a reason why there is a `MachineLICM` and `EarlyMachineLICM` pass. But they don't really matter because`PreRegAlloc` is redefined anyway.

This diff is more of a band-aid to make MIR tests easy to write, because the MIRParser considers MIR as SSA if it has absolutely no vreg, which is unfortunate.